data/src/main/java/me/androidbox/data/worker_manager/TaskCreateSynchronizerWorker.kt

```diff
@@ -0,0 +1,150 @@
+
+                reminderIds.map { id ->
+                    reminderDao.getReminderById(id)
+                }
+            }
+
+            updatedReminders = async {
+                val reminderIds = reminderDao.getAllRemindersBySyncType(SyncAgendaType.UPDATE)
+
+                reminderIds.map { id ->
+                    reminderDao.getReminderById(id)
+                }
+            }
+        }
+
+        /** TODO I am using supervisor scope to launch so if one fails the other will still run
+         *  However, will the worker return failure if a single one fails even though some failed?
+         *  My checkResult will run everything in this block */
+        val responseState = checkResult {
+            supervisorScope {
+                val createTaskJob = launch {
+                    createdTasks.await().map { taskEntity ->
+                        /* Create the task on remote */
+                        taskService.createTask(taskEntity.toTaskDto())
+                        /* If success will run this delete - else if loop next next one */
+                        taskDao.deleteSyncTaskById(taskEntity.id)
+                    }
+                }
+
+                val updateTaskJob = launch {
+                    updatedTasks.await().forEach { taskEntity ->
+                        taskService.updateTask(taskEntity.toTaskDto())
+                        taskDao.deleteSyncTaskById(taskEntity.id)
+                    }
+                }
+
+                val createReminderJob = launch {
+                    createdReminders.await().forEach { reminderEntity ->
+                        reminderService.createReminder(reminderEntity.toReminderDto())
+                        reminderDao.deleteSyncReminderById(reminderEntity.id)
+                    }
+                }
+
+                val updateReminderJob = launch {
+                    updatedReminders.await().forEach { reminderEntity ->
+                        reminderService.updateReminder(reminderEntity.toReminderDto())
+                        reminderDao.deleteReminderById(reminderEntity.id)
+                    }
+                }
+
+                /** TODO Waiting for all jobs to complete */
+                createTaskJob.join()
+                updateTaskJob.join()
+                createReminderJob.join()
+                updateReminderJob.join()
+            }
+        }
+
+        return responseState.fold(
+            onSuccess = {
+                /** TODO Success we can delete all these from the sync table
+                 *  However, I don't think I need to do this now if
+                 *  1) I am already deleting the items individually when create/updated
+                 *  2) If the worker is success, but i.e. some reminders failed to create it will empty them
+                 *     and when the worker runs again, there will be nothing to fetch.
+                 *  */
```

Comment:
You are already removing them one by one, so this should be unnecessary


" 2) If the worker is success, but i.e. some reminders failed to create it will empty them and when the worker runs again, there will be nothing to fetch."
In which case would the worker be success, but the reminder failed to create? If it fails to create, it should return a Result.retry()

Reply:
@MKiperszmid I have just add some snippet here as I want to clarify.

I have supervisor scope that is wrapped in checkResult block. Supervisor won't cancel any children. And at the end I check the responseState if it is success or failure. But if 1 launch was to fail, I am not sure how to handle that in this current implementation

```
       val responseState = checkResult {
            supervisorScope {
                val createTaskJob = launch {
                    createdTasks.await().forEach { taskEntity ->
                        /* delete if success */
                    }
                }

                val updateTaskJob = launch {
                    updatedTasks.await().forEach { taskEntity ->
                        /* delete if success */
                    }
                }

                val createReminderJob = launch {
                    createdReminders.await().forEach { reminderEntity ->
                        /* delete if success */
                    }
                }

                val updateReminderJob = launch {
                    updatedReminders.await().forEach { reminderEntity ->
                        /* delete if success */
                    }
                }

                /* do all the joins */
            }
        }

        return responseState.fold(
            onSuccess = {
                Result.success()
            },
            onFailure = {
                Result.retry()
            }
        )
    }
```

Reply:
One quick solution could be to have a boolean like:

var workerFailed = false

And if there's any fails, set it to true

And then just
```
if (workerFailed) {
    return Result.retry()
} else {
    return Result.success()
}
```